### PR TITLE
Add sha384 and sha512 to SRI

### PR DESCRIPTION
**Pull request by danielfain**
This is for issue #11123

## Git commit checklist

* [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by MattIPv4** [comment]
Hey, @danielfain thanks for jumping on this super quick!

Per the suggestion of @drewfreyling in #13384 I think it might be better to just generate the SHA-512. However, I think it'd be great to leave the existing code you've written in place for 256/384 in the event we decide to ever use them, just comment it out for now :)

**Comment by danielfain** [comment]
Hey, @MattIPv4 this is my first time contributing to an open source project so I'm not quite sure the best way to tweak it. Should I just make another commit on my branch with the aforementioned fixes or should I rebase both commits into 1?

Thanks!

**Comment by MattIPv4** [comment]
@danielfain If you're happy with rebasing, a single commit would be nicer :)

File: build/packages.json.js [hunk 1]
```diff
@@ -213,14 +213,32 @@ glob('ajax/libs/*/package.json', function (error, matches) {
...
+              if (file256SRI !== '' && file384SRI !== '' && file512SRI !== '') {
+                libSri[temp.files[i]] = 'sha256-' + file256SRI + 
+                  ' sha384-' + file384SRI + 
+                  ' sha512-' + file512SRI;
+              }
+              */
+              if (file512SRI !== '') {
+                libSri[temp.files[i]] = 'sha512-' + file512SRI;
               }
-
```
Comment (PeterDaveHello) [inline build/packages.json.js]: this line change is not related

**Comment by MattIPv4** [comment]
@xtuc What are your thoughts on this?

**Comment by xtuc** [comment]
That's nice. The right place is now in https://github.com/cdnjs/tools/blob/master/openssl/sri.go#L10.

Also I'm wondering if this should apply to the existing assets on the cdn, because it will take a fair amour of time to generate.

My understanding is some non-browsers could rely on the hash via the API as well, shouldn't we support both?

**Comment by MattIPv4** [comment]
Yeah, that was the main issue when this was originally suggested - do we just jump to 512? Do we provide 256, 384 & 512 in a single string? Do we default to 512 and provide 256/384 via additional SRI attributes in the API? Do we generate the new hashes for all lib files? Do we only generate for new files once implemented?

**Comment by MattIPv4** [comment]
To me, either of these two makes sense:

Jump to 512 and only generate it going forward. Old assets will still just use 256 and anything new will just use 512.

Providing 512 by default with 256 & 384 available as additional attrs. And, we generate the hashes for all files on cdnjs, so that the API always has correctly populated SRI data.

**Comment by rarkins** [comment]
Hi, we are mostly done implementing cdnjs updating in [Renovate Bot](https://github.com/renovatebot/renovate). The problem I'm seeing is that we are "downgrading" the sha strength for people who have `sha384` or `sha512`. Although the theory behind "only sha512" seems fine to me, I'd still prefer that all of the hashes are made available in the API so that we can match with the one they currently have and cause least "friction'.

Here's an example of a PR today when we use the SRI from the cdnjs API:

image

**Comment by MattIPv4** [comment]
SRI generation is no longer handled through this script. Closing.
We can look at supporting stronger SRI hashes through the new scripting.

**Comment by rarkins** [comment]
@MattIPv4 do you have any issue or link I can follow?

**Comment by MattIPv4** [comment]
I would continue tracking https://github.com/cdnjs/cdnjs/issues/11123, we'll post updates there as we go.
